### ISSUE TITLE
@ExportDecoratedItems stopped working for properties

### ISSUE THREAD

**Issue by rubenlg** [angular/tsickle#558]
`@ExportDecoratedItems` used to work on properties, but not anymore. In fact, the existing unit test is no longer testing that properties work. I believe this behavior changed in #450. At least, that PR changed the test to no longer verify that the `exportMe` property is indeed exported.

@drjanitor, @evmar, was that intended? If so, perhaps we should change the unit test to not create the impression that you can use it on properties like `exportMe`.

**Comment by drjanitor** [comment]
https://github.com/angular/tsickle/blob/master/src/tsickle.ts#L821

check what happens to the code around that line. we are definitely /trying/
to add that jsdoc.

**Comment by drjanitor** [comment]
checked the test, and your bug report seems to be missing some crucial
information.

tsickle correctly emits the @export annotation: see
https://github.com/angular/tsickle/blob/master/test_files/exporting_decorator/exporting.tsickle.ts#L25

the problem comes with the fact that property declarations are moved to
MyClass_tsickle_Closure_declarations, and the moving part seems to eat
@export:

https://github.com/angular/tsickle/blob/master/test_files/exporting_decorator/exporting.js#L103

in short, whatever moves property declarations down there ought to respect
the existing jsdoc tags.

there is no difference between what we do and manually specifying @export
in the .ts file, which seems to point at the fact that the annotation just
gets dropped on the floor somewhere.

**Comment by drjanitor** [comment]
https://github.com/angular/tsickle/blob/master/src/tsickle.ts#L1442

possible problem: this line operates on the original .ts file, not on
.tsickle.ts.
we may need to add the hasExportingDecorator() call here too.

**Comment by drjanitor** [comment]
[drive-by: the comment at
https://github.com/angular/tsickle/blob/master/src/tsickle.ts#L1326
mentions _tsickle_typeAnnotationsHelper, which I assume is the old name of
_tsickle_Closure_declarations.]

**Comment by rubenlg** [comment]
Thanks!
